orca-webhook/src/test/groovy/com/netflix/spinnaker/orca/webhook/WebhookServiceSpec.groovy

```diff
@@ -89,17 +89,29 @@ class WebhookServiceSpec extends Specification {
   }
 
   def "Status endpoint is being called"() {
+
     expect:
-    server.expect(requestTo("https://localhost/v1/status/123"))
+    def responseActions = server.expect(requestTo("https://localhost/v1/status/123"))
       .andExpect(method(HttpMethod.GET))
-      .andRespond(withSuccess('["element1", 123, false]', MediaType.APPLICATION_JSON))
+
+      if(customHeaders){
+        customHeaders.each {k, v -> responseActions.andExpect(header(k, v))}
+      }
+
+      responseActions.andRespond(withSuccess('["element1", 123, false]', MediaType.APPLICATION_JSON))
 
     when:
-    def responseEntity = webhookService.getStatus("https://localhost/v1/status/123")
+    def responseEntity = webhookService.getStatus("https://localhost/v1/status/123", [Authorization: "Basic password"])
 
     then:
     server.verify()
     responseEntity.statusCode == HttpStatus.OK
     responseEntity.body == ["element1", 123, false]
+
+    where:
+    customHeaders                     | _
+    [Authorization: "Basic password"] | _
+    [:]                               | _
+    null                              | _
```

Comment:
Might be better to write this with a data pipe (good for when you don't have an expected value you're testing:
```groovy
where:
customHeaders << [ [Authorization: "Basic password"], [:], null ]
```
Also want to include an `@Unroll` annotation at the top of this spec, so runs are reported separately by the test runner.